examples/ecs/system_sets.rs

```diff
@@ -0,0 +1,157 @@
+///     \--> movement               (via label PhysicsSystem::Movement)
+/// PostPhysics                 (Criteria: Resource `done` is false)
+///     \--> collision || sfx
+/// Exit                        (Criteria: Resource `done` is true)
+///     \--> exit
+/// ```
+///
+/// The `Physics` label represents a [SystemSet] containing two systems.
+/// This set's criteria is to stop after a second has elapsed.
+/// The two systems (update_velocity, movement) runs in a specified order.
+///
+/// Another label `PostPhysics` uses run criteria to only run after `Physics` has finished.
+/// This set's criteria is to run only when _not done_, as specified via a resource.
+/// The two systems here (collision, sfx) are not specified to run in any order, and the actual
+/// ordering can then change between invocations.
+///
+/// Lastly a system with run criterion _done_ is used to exit the app.
+/// ```
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .init_resource::<Done>()
+        .add_system_set(
+            SystemSet::new()
+                // This label is added to all systems in this set.
+                // The label can then be referred to elsewhere (other sets).
+                .label(Physics)
+                .with_system(
+                    update_velocity
+                        .system()
+                        // Only applied to the `update_velocity` system
+                        .label(PhysicsSystem::UpdateVelocity),
+                )
+                .with_system(
+                    movement
+                        .system()
+                        // Only applied to the `movement` system
+                        .label(PhysicsSystem::Movement)
+                        // Enforce order within this system by specifying this
+                        .after(PhysicsSystem::UpdateVelocity),
+                )
+                // This criteria ensures this whole system set only runs when this system's
+                // output says so (ShouldRun::Yes)
+                .with_run_criteria(run_for_a_second.system()),
+        )
+        .add_system_set(
+            SystemSet::new()
+                .label(PostPhysics)
+                // `collision` and `sfx` are not ordered with respect to
+                // each other, and may run in any order
+                .with_system(collision.system())
+                .with_system(sfx.system())
+                // This whole set runs after `Physics` (which in this case is a label for
+                // another set).
+                // There is also `.before(..)`.
+                .after(Physics)
+                // This shows that we can modify existing run criteria results.
+                // Here we create a _not done_ criteria by piping the output of
+                // the `done` system and inverting the output.
+                .with_run_criteria(RunCriteria::pipe("done", inverse.system())),
```

Comment:
The `"done"` string realy confused me here, as i thought it was somehow finding the correct function through a string alone, and not a label.
To reduce this confusion, I suggest to use another Type that derives SystemLabel instead.

Reply:
I varied names slightly to differentiate better.

I wanted to have an example of using the string literal for creating a `SystemLabel` since my intuition says there are probably users who would like to uses labels but not create extra types for it.

Hopefully the new names and added comment makes things clearer.